src/main/java/org/minions/devfund/carlosorellana/Squarelotron.java

```diff
@@ -0,0 +1,276 @@
+                    this.squarelotron[size - 1 - positionX][positionY];
+        }
+
+
+        if (valC && valD) {
+            squarelotronTemporal.squarelotron[positionX][positionY] =
+                    this.squarelotron[size - 1 - positionX][positionY];
+        }
+
+        return squarelotronTemporal;
+    }
+
+
+        /**
+         * Modifies the matrix - move the number diagonally in the Matrix.
+         *
+         * @param ring - Number the ring affected.
+         * @return New instance of Squarelotron
+         */
+
+    public Squarelotron mainDiagonalFlip(final int ring) {
+
+        if (ring > this.getMaxRings() || ring < 1) {
+            throw new NumberFormatException();
+        }
+
+        Squarelotron squarelotronTemporal = new Squarelotron(this.size);
+
+        int first = ring - 1; // 0
+        int last = size - ring; // 3
+
+
+        for (int i = 0; i <= size - 1; i++) {
+            for (int j = 0; j <= size - 1; j++) {
+                squarelotronTemporal = createNewMatrixToMainDiagonalFlip(i, j, first, last, squarelotronTemporal);
+            }
+        }
+        return squarelotronTemporal;
+    }
+
+    /**
+     * Creates a new matrix to Main Diagonal Flip values.
+     *
+     * @param positionX            - Position X.
+     * @param positionY            - Position Y.
+     * @param first                - Value get to (ring - 1).
+     * @param last                 - Value get to (size - ring).
+     * @param squarelotronTemporal - New Squarelotron Object.
+     * @return Instance of Squarelotron.
+     */
+    private Squarelotron createNewMatrixToMainDiagonalFlip(final int positionX, final int positionY,
+                                                           final int first, final int last,
+                                                           final Squarelotron squarelotronTemporal) {
+        boolean valA = positionX == first || positionX == last;
+        boolean valB = positionY >= first && positionY <= last;
+        boolean valC = positionX > first && positionX < last;
+        boolean valD = positionY == first || positionY == last;
+
+        if (valA && valB) {
+            squarelotronTemporal.squarelotron[positionY][positionX] = this.squarelotron[positionX][positionY];
+        }
+
+        if (valC && valD) {
+            squarelotronTemporal.squarelotron[positionY][positionX] = this.squarelotron[positionX][positionY];
+        }
+        return squarelotronTemporal;
+    }
+
+    /**
+     * Rotates Matrix 90 grades positive or negative position.
+     *
+     * @param numberOfTurns Number or rotations.
+     */
+    public void rotateRight(final int numberOfTurns) {
+
+        int turns = numberOfTurns;
+        int ring;
+        boolean right = true;
+        int maxRing = getMaxRings();
+
+        // New Temporal Squarelotron
+        int[][] temporalSquarelotron = new int[size][size];
+        temporalSquarelotron = duplicateSquarelotronMatrix(temporalSquarelotron);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment; "temporalSquarelotron" already holds the assigned value along all execution paths. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4165)